### DISCUSSION TITLE
[@types/node] missing globals (e.g. URL)

### DISCUSSION THREAD

**Discussion by tjwelde** [DefinitelyTyped/DefinitelyTyped#57055] [Issues with a @types package]
URL and URLSearchParams are available on the global object since node v10.0.0, but aren't included in the types.

Docs on nodejs: 
- https://nodejs.org/api/globals.html#url
- https://nodejs.org/api/globals.html#urlsearchparams

I didn't try it, but it seems like other global properties are missing, too, like `WebAssembly`, `TextEncoder`, `TextDecoder`.
Is there a reasoning behind not including this?

[a comment by an automated account is not shown]

**Comment by peterblazejewicz** [comment]
This is probably due to recurring problems with clean separation of types resolutions when using node (have you ever tried to use `setTimeout` in a project using both standard dom and referencing node in tooling?) and problem of Node reimplementing standard apis.
The `url` module contains those symbols, globals usually resolve here to standard library and there is ongoing effort to put standard WHATWG api

**Comment by yume-chan** [comment]
`TextEncoder` and `TextDecoder` are available on global from Node.js 11.0.0, docs:

* https://developer.mozilla.org/en-US/docs/Web/API/TextDecoder#browser_compatibility 
* https://nodejs.org/api/util.html#new-textdecoderencoding-options

colliding with `lib: ["dom"]` is painfully, but that's usually caused by some nested dependencies incorrectly imported `@types/node` into my package, that's another problem.

**Comment by thw0rted** [comment]
I'm trying to remove `lib: ["dom"]` in https://github.com/DefinitelyTyped/DefinitelyTyped/pull/59905.  One test used global `WebAssembly` but didn't actually need it, no tests used `TextEncoder` or `TestDecoder`.  If I actually land this PR, maybe somebody here would like to step up and add those?
